plugin/src/main/java/io/jenkins/plugins/forensics/git/reference/GitReferenceRecorder.java

```diff
@@ -70,14 +74,57 @@ public void setSkipUnknownCommits(final boolean skipUnknownCommits) {
     public boolean isSkipUnknownCommits() {
         return skipUnknownCommits;
     }
+    
+    /**
+     * Determines the reference build selection strategy, if
+     *  BEST_MATCH_BUILD: default reference point will be used
+     *  PARENT_COMMIT_BUILD: a reference build, which has built the parent commit of the current built revision
+     *  
+     * @param referenceBuildFinderStrategy         
+     */
+    @DataBoundSetter
+    public void setReferenceBuildFinderStrategy(ReferenceBuildFinderStrategy referenceBuildFinderStrategy) {
+        this.referenceBuildFinderStrategy = referenceBuildFinderStrategy;
+    }
+    
+    public ReferenceBuildFinderStrategy getReferenceBuildFinderStrategy() {
+        return referenceBuildFinderStrategy;
+    }
+    
+    /**
+     * Tries to find a reference build, which built the parent commit of the current build
+     * @param owner
+     *          the current build
+     * @param parentCommit
+     *          the parentCommit-SHA1, which was attached to the current build
+     *          
+     * @return reference build, if exists
+     */
+    private Optional<Run<?, ?>> findParentReferenceBuildFromCommit(final Run<?, ?> owner, 
+            final String parentCommit) {
+        for(Run<?, ?> run = owner.getPreviousBuild(); run != null; run = run.getPreviousBuild()) {
+            GitCommitsRecord record = run.getAction(GitCommitsRecord.class);
+            if(record != null && parentCommit.equals(record.getLatestCommit())) {
+                return Optional.of(run);
+            }
+        }
+        return Optional.empty();
+    }
 
     @Override
     protected Optional<Run<?, ?>> find(final Run<?, ?> owner, final Run<?, ?> lastCompletedBuildOfReferenceJob) {
         GitCommitsRecord thisCommit = owner.getAction(GitCommitsRecord.class);
-        GitCommitsRecord referenceCommit = lastCompletedBuildOfReferenceJob.getAction(GitCommitsRecord.class);
+        
+        if(referenceBuildFinderStrategy == ReferenceBuildFinderStrategy.BEST_MATCH_BUILD) {
+            GitCommitsRecord referenceCommit = lastCompletedBuildOfReferenceJob.getAction(GitCommitsRecord.class);
 
-        return thisCommit.getReferencePoint(referenceCommit, getMaxCommits(), isSkipUnknownCommits());
-    }
+            return thisCommit.getReferencePoint(referenceCommit, getMaxCommits(), isSkipUnknownCommits());
+        } else {
+            String parentCommit = thisCommit.getParentCommit();
+            
+            return findParentReferenceBuildFromCommit(owner, parentCommit);
```

Comment:
This depends a little bit on the answer of the comment above. But in the end it would be nicer if we can use inheritance rather than an `if` for the different strategies.